### ISSUE TITLE
sync_capsule_repos_to_upgrade task is quite dependant on nailgun

### ISSUE THREAD

**Issue by ntkathole** [SatelliteQE/satellite6-upgrade#282]
https://github.com/SatelliteQE/satellite6-upgrade/blob/6cfdfa89bf7fb1d8f226d5d2deaf7a9a2c40adeb/upgrade/helpers/tasks.py#L72
Capsule repo sync uses nailgun which has a lot of variations in code/attributes between satellite versions.
The following issues need to fix or let's move back this task to CLI based.

- Satellite6-upgrader jobs cant pick up nailgun version according to satellite version. (upgrade using this job is blocked. This job is useful for creating upgrade templates)
- Zstream upgrade in pipeline needs to do manual changes. As nailgun is picked up according to "TO version". Thus, for 6.4 to 6.4.z upgrade, 6.4.z branch is used but actual nailgun code to be used will be in master.

**Comment by jyejare** [comment]
> [satellite6-upgrade/upgrade/helpers/tasks.py](https://github.com/SatelliteQE/satellite6-upgrade/blob/6cfdfa89bf7fb1d8f226d5d2deaf7a9a2c40adeb/upgrade/helpers/tasks.py#L72)
> 
> Line 72 in [6cfdfa8](/SatelliteQE/satellite6-upgrade/commit/6cfdfa89bf7fb1d8f226d5d2deaf7a9a2c40adeb)
>  def sync_capsule_repos_to_upgrade(capsules): 
> 
> Capsule repo sync uses nailgun which has a lot of variations in code/attributes between satellite versions.
> The following issues need to fix or let's move back this task to CLI based.
> 
>     * Satellite6-upgrader jobs cant pick up nailgun version according to satellite version. (upgrade using this job is blocked. This job is useful for creating upgrade templates)
True, satellite6-upgrader needs updates with the latest changes, especially wrt nailgun. We can follow TO_VERSION and FROM_VERSION for installing nailgun and this strategy will work for all the versions except LATEST_SAT_VERSION, PREVIOUS_VERSION.
e.g if TO_VERSION='6.5' during upgrade, then we cant install nailgun#6.5.z as its not available. The available is master.
So there are two ways to fix it:
1. Wait for nailgun to implement something like this Sat 6.5(latest) -> nailgun#6.5.z
2. Keep a mapping of nailgun master and satellite somewhere in ci shell files or in the upgrade. I vote for ci.
>     * Zstream upgrade in pipeline needs to do manual changes. As nailgun is picked up according to "TO version". Thus, for 6.4 to 6.4.z upgrade, 6.4.z branch is used but actual nailgun code to be used will be in master.
As discussed, Implement in ci shell file like:
if ZSTREAM_UPGRADE:
       ```install nailgun of FROM_VERSION before and after upgrade as well```

**Comment by ntkathole** [comment]
@jyejare This is still an issue.

**Comment by jyejare** [comment]
Do we implement the above comment?
